Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -0,0 +1,249 @@
+                "$weightedAttenuationHigh + $defaultBucketOffset)"
+        Timber.v(attenuationStrings)
+
+        val weightedAttenuationDuration =
+            weightedAttenuationLow
+                .plus(weightedAttenuationMid)
+                .plus(weightedAttenuationHigh)
+                .plus(defaultBucketOffset)
+
+        Timber.v("Formula used: ($maximumRiskScore / $normalizationDivisor) * $weightedAttenuationDuration")
+
+        val riskScore = (maximumRiskScore / normalizationDivisor) * weightedAttenuationDuration
+
+        return round(riskScore.times(DECIMAL_MULTIPLIER)).div(DECIMAL_MULTIPLIER)
+    }
+
+    @VisibleForTesting
+    internal fun Int.capped() =
+        if (this > TimeVariables.getMaxAttenuationDuration()) {
+            TimeVariables.getMaxAttenuationDuration()
+        } else {
+            this
+        }
+
+    @VisibleForTesting
+    internal fun withinDefinedLevelThreshold(riskScore: Double, min: Int, max: Int) =
+        riskScore >= min && riskScore <= max
+
+    /**
+     * Updates the Risk Level Score in the repository with the calculated Risk Level
+     *
+     * @param riskLevel
+     */
+    @VisibleForTesting
+    internal fun updateRiskLevelScore(riskLevel: RiskLevel) {
+        val lastCalculatedScore = RiskLevelRepository.getLastCalculatedScore()
+        if (RiskLevel.riskLevelChangedBetweenLowAndHigh(
+                lastCalculatedScore,
+                riskLevel
+            ) && !LocalData.submissionWasSuccessful()
+        ) {
+            NotificationHelper.sendNotification(
+                CoronaWarnApplication.getAppContext().getString(R.string.notification_body),
+                NotificationCompat.PRIORITY_HIGH
+            )
+        }
+        if (lastCalculatedScore.raw == RiskLevelConstants.INCREASED_RISK &&
+            riskLevel.raw == RiskLevelConstants.LOW_LEVEL_RISK) {
+            LocalData.isUserToBeNotifiedOfLoweredRiskLevel = true
+        }
+        RiskLevelRepository.setRiskLevelScore(riskLevel)
+    }
+
+    /**
+     * If there is no persisted exposure summary we try to get a new one with the last persisted
+     * Google API token that was used in the [de.rki.coronawarnapp.transaction.RetrieveDiagnosisKeysTransaction]
+     *
+     * @return a exposure summary from the Google Exposure Notification API
+     */
+    private suspend fun getNewExposureSummary(): ExposureSummary {
+        val googleToken = LocalData.googleApiToken()
+            ?: throw RiskLevelCalculationException(IllegalStateException("Exposure summary is not persisted"))
+
+        val exposureSummary =
+            InternalExposureNotificationClient.asyncGetExposureSummary(googleToken)
+
+        return exposureSummary.also {
+            Timber.tag(TAG)
+                .v("Generated new exposure summary with $googleToken")
+        }
+    }
+
+    companion object {
+        private var TAG = DefaultRiskLevels::class.simpleName
```

Comment:
TAG should be final (val) I think..
I also like private val TAG = DefaultRiskLevels::class.java.simpleName very much, because you get a String instead of String?, but please, don't change it if you don't want (just a personal note)